### PR TITLE
prioritize the result with the image

### PR DESCRIPTION
**Pull request by adamrocker**
The result of a keyboard with an image will be listed higher.

スクリーンショット 2021-07-30 13 51 55

スクリーンショット 2021-07-30 13 59 29

### CODE REVIEW

**Review by yoichiro — CHANGES_REQUESTED** [review]
@adamrocker The search result is sorted in [here](https://github.com/remap-keys/remap/blob/ea679a8f0629f349f5653898f0e9b6d95a0d7fc8/src/actions/storage.action.ts#L800-L808). How about adding the new sort condition of each result has an image or not to the existing logic above?

**Comment by adamrocker** [comment]
@yoichiro Thanks for your help. I fixed them. Plz check my new commit.
スクリーンショット 2021-07-30 16 09 14

スクリーンショット 2021-07-30 16 09 35

File: src/actions/storage.action.ts [hunk 1]
```diff
@@ -793,17 +793,22 @@ export const storageActionsThunk = {
...
+        let count = doc.features.reduce<number>((result, feature) => {
+          return features!.includes(feature) ? result + 2 : result; // added 2 points because matching features is more important than the kbd has an image
         }, 0);
+        if (doc.imageUrl) {
+          count += 1; // sort higher if the keyboard has an image
+        }
+        return count;
       };
-      const sortedSearchResult = definitionDocs.slice().sort((a, b) => {
+      const sortedSearchResult = Array.from(definitionDocs).sort((a, b) => {
```
Comment (yoichiro) [inline src/actions/storage.action.ts:804]: 👍

File: src/actions/storage.action.ts [hunk 2]
```diff
@@ -793,17 +793,22 @@ export const storageActionsThunk = {
       const matchedFeaturesCount = (
         doc: IKeyboardDefinitionDocument
       ): number => {
-        return doc.features.reduce<number>((result, feature) => {
-          return features!.includes(feature) ? result + 1 : result;
+        let count = doc.features.reduce<number>((result, feature) => {
+          return features!.includes(feature) ? result + 2 : result; // added 2 points because matching features is more important than the kbd has an image
```
Comment (yoichiro) [inline src/actions/storage.action.ts:797]: 👍

**Review by yoichiro — APPROVED** [review]
@adamrocker LGTM! Thanks!

**Pull request merged**